### PR TITLE
CFY-5970-make-postgresql-idempotent

### PR DESCRIPTION
**Pull request by dusking**

### CODE REVIEW

File: components/postgresql/scripts/config.sh [hunk 1]
```diff
@@ -6,7 +6,8 @@ function update_ps_configuration() {
     ctx logger info "Updating Configuration..."
     pg_hba="/var/lib/pgsql/9.5/data/pg_hba.conf"
     sudo cp $pg_hba $pg_hba.backup
-    awk_replace="/^host/{gsub(/ident/, \"md5\")}; {print}"
+    ctx logger info "Going to modify $pg_hba"
```
Comment (nir0s) [inline components/postgresql/scripts/config.sh]: debug

Comment (nir0s) [inline components/postgresql/scripts/config.sh]: debug, Modifying ...

File: components/postgresql/scripts/create.py [hunk 2]
```diff
@@ -44,7 +45,11 @@ def _install_postgresql():
 def _init_postgresql():
     ctx.logger.info('Init PostreSQL DATA folder...')
```
Comment (nir0s) [inline components/postgresql/scripts/create.py]: Initializing PostgreSQL data folder...

File: components/postgresql/scripts/create.py [hunk 3]
```diff
@@ -44,7 +45,11 @@ def _install_postgresql():
...
     postgresql95_setup = '/usr/pgsql-9.5/bin/postgresql95-setup'
-    utils.sudo(command=[postgresql95_setup, 'initdb'])
+    try:
+        utils.sudo(command=[postgresql95_setup, 'initdb'])
+    except Exception:
+        ctx.logger.info('PostreSQL DATA folder already been init...')
```
Comment (nir0s) [inline components/postgresql/scripts/create.py]: debug

File: components/postgresql/scripts/create.py [hunk 4]
```diff
@@ -54,7 +59,8 @@ def _init_postgresql():
     ps_95_logs_path = "/var/lib/pgsql/9.5/data/pg_log"
     ps_logs_path = "/var/log/cloudify/postgresql"
     utils.mkdir(ps_logs_path)
-    utils.ln(source=ps_95_logs_path, target=ps_logs_path, params='-s')
+    if not os.path.exists(ps_95_logs_path):
```
Comment (nir0s) [inline components/postgresql/scripts/create.py]: isdir

File: components/postgresql/scripts/start.py [hunk 5]
```diff
@@ -26,23 +28,53 @@ def _start_postgres():
 def _create_default_db(db_name, username, password):
     ctx.logger.info('Creating default postgresql database: {0}...'.format(
         db_name))
-    # TODO: join...
-    ps_config_dir = 'components/postgresql/config'
-    ps_config_source = join(ps_config_dir, 'create_default_db.sh')
-    # TODO: Replace with tempfile.gettempdir() and join
-    ps_config_destination = '/tmp/create_default_db.sh'
+    ps_config_source = join('components/postgresql/config',
```
Comment (nir0s) [inline components/postgresql/scripts/start.py]: this join doesn't matter

File: components/postgresql/scripts/start.py [hunk 6]
```diff
@@ -26,23 +28,53 @@ def _start_postgres():
...
+        host=host,
+        port=postgresql_default_port,
+        db_name=db_name,
+        user=username,
+        password=password
+    )
+    # .pgpass file used by mgmtworker in snapshot workflow,
+    # and need to be under th home directory of the user who run the snapshot
+    # (currently root)
+    if os.path.exists(pgpass_path):
```
Comment (nir0s) [inline components/postgresql/scripts/start.py]: isddir

File: components/postgresql/scripts/start.py [hunk 7]
```diff
@@ -26,23 +28,53 @@ def _start_postgres():
...
+        ctx.logger.info('Deleting {0} file, going to create it..'.format(
```
Comment (nir0s) [inline components/postgresql/scripts/start.py]: debug

File: components/postgresql/scripts/start.py [hunk 8]
```diff
@@ -26,23 +28,53 @@ def _start_postgres():
...
+            pgpass_path
+        ))
+        os.remove(pgpass_path)
+    with os.fdopen(os.open(pgpass_path, os.O_WRONLY | os.O_CREAT, 0600), 'w')\
+            as pgpass_file:
+        pgpass_file.write(pgpass_content)
+        ctx.logger.debug('Postgresql pass file {0} created..'.format(
```
Comment (nir0s) [inline components/postgresql/scripts/start.py]: remove ..

File: components/postgresql/scripts/start.py [hunk 9]
```diff
@@ -26,23 +28,52 @@ def _start_postgres():
...
+        port=postgresql_default_port,
+        db_name=db_name,
+        user=username,
+        password=password
+    )
+    # .pgpass file used by mgmtworker in snapshot workflow,
+    # and need to be under th home directory of the user who run the snapshot
+    # (currently root)
+    if os.path.isfile(pgpass_path):
+        ctx.logger.debug('Deleting {0} file, going to create it..'.format(
```
Comment (nir0s) [inline components/postgresql/scripts/start.py]: remove going to create..

**Comment by dusking** [comment]
Reviewed by NrC

**Pull request merged**